refactor: Rename `Huddle` to `DirectMessageGroup` in the model class.

roanster007 opened the pull request:
<!-- Describe your pull request here.-->

Commit 1: Renames `Huddle` django model to `DirectMessageGroup`, with  `db_table="zerver_huddle"`, thus preserving the table name.



Part of #28640

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on zerver/migrations/0542_rename_huddle_directmessagegroup_and_more.py:
```diff
@@ -0,0 +1,20 @@
+            name="directmessagegroup",
+            table="zerver_huddle",
+        ),
+    ]
```
Does this migration generate any SQL via `manage.py sqlmigrate`? I worry that it might do something crazy, which could be avoided with advanced features like `state_operations`.

roanster007 replied:
Updated! `manage.py sqlmigrate`:
```
BEGIN;
--
-- Custom state/database change combination
--
-- (no-op)
COMMIT;
```

timabbott replied:
Next time post what the original `sqlmigrate` output was too to just fully explain the trade-off and what changed; I was curious enough that I ended up checking myself.

timabbott commented on zerver/models/recipients.py:
```diff
@@ -112,44 +114,49 @@ def bulk_get_direct_message_group_user_ids(recipient_ids: List[int]) -> Dict[int
     recipient = models.ForeignKey(Recipient, null=True, on_delete=models.SET_NULL)
 
+    class Meta:
+        db_table = "zerver_huddle"
```
Let's add a comment explaining that this is the old name and should change.

roanster007 replied:
Added 
```
# TODO: As a part of migration of Huddle to DirectMessageGroup, the
# "zerver_huddle" table needs to be renamed to "zerver_directmessagegroup".
```

timabbott commented on zerver/migrations/0543_rename_huddle_hash_directmessagegroup_direct_message_group_hash.py:
```diff
@@ -0,0 +1,17 @@
+            old_name="huddle_hash",
+            new_name="direct_message_group_hash",
+        ),
+    ]
```
I think if we're doing a disruptive database migration like this, one should rename the table too. But I think it's better to defer this by just dropping this commit; I don't want to manage this class of disruptive database migration for 9.0.

roanster007 replied:
dropped the commit!

roanster007 commented:
@timabbott
>I think if we're doing a disruptive database migration like this, one should rename the table too. But I think it's better to defer this by just dropping this commit; I don't want to manage this class of disruptive database migration for 9.0.

removed the commit changing the `DirectMessageGroup` field.

>Does this migration generate any SQL via manage.py sqlmigrate? I worry that it might do something crazy, which could be avoided with advanced features like state_operations.

Updated to use `state_operations`. So `./manage.py sqlmigrate` is a no-op now.

timabbott commented:
Looks good, renumbered the migration and merged, thanks @roanster007!
